src/choreo/test_choreo.c

```diff
@@ -0,0 +1,1163 @@
+#include "tower/fd_tower.h"
+#include "forks/fd_forks.h"
+#include "ghost/fd_ghost.h"
+#include "../flamenco/runtime/program/fd_vote_program.h"
+#include "../flamenco/runtime/program/fd_vote_program.c"
+#include "../flamenco/runtime/fd_txn_account.c"
+#include "../funk/fd_funk_filemap.h"
+#include "../util/wksp/fd_wksp.h"
+
+
+fd_tower_t *
+make_tower(fd_wksp_t * wksp, size_t n, ...) {
+  void * tower_mem = fd_wksp_alloc_laddr( wksp, fd_tower_align(), fd_tower_footprint(), 6UL );
```

Comment:
make the mem passed in, so caller is responsible for both allocing and freeing. otherwise caller has to remember to free something they didn't allocate

alternatively you can just put the mem in global scope similar to how kevin does it here
https://github.com/firedancer-io/firedancer/blob/main/src/util/tmpl/test_treap.c#L58-L60

so you don't have to pass it around